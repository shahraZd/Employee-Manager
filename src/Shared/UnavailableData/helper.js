import React, { Fragment } from 'react'
import { isString, isArray } from 'lodash'

export const renderErrorMessage = message => {
  if (isString(message)) {
    return message
  }
  if (isArray(message)) {
    return message.map(errorMessage => <div>{errorMessage}</div>)
  }
  if (typeof message === 'object') {
    return message
  }
  return <Fragment />
}
