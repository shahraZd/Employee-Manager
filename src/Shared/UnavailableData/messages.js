/*
 * UnavailableData Messages
 *
 * This contains all the text for the UnavailableData component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.UnavailableData'

export default defineMessages({
  trackFailureMessage: {
    id: `${scope}.trackFailureMessage`,
    defaultMessage: 'No data available for now',
  },
  backDashboard: {
    id: `${scope}.backDashboard`,
    defaultMessage: 'Back to Dashboard',
  },
})
