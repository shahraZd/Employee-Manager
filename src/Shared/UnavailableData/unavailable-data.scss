.unavailable-data {
  height: 80%;

  .tracks-failure-message {
    text-align: center;

    .unavailable-data-icon {
      margin-bottom: 30px;

      .anticon {
        font-size: 180px !important;
        color: #dfdfdf !important;
      }
    }

    .unavailable-data-message {
      margin-bottom: 40px;
      font-size: 23px;
      color: #393c40;
      line-height: 17px;
    }

    .unavailable-data-btn {
      width: 160px;
      height: 50px;
      background-color: #170ff0 !important;
      border-radius: 3px;
      font-size: 16px;
      color: white;
      font-weight: normal;
      padding: 0;
      border-color: transparent !important;
    }
  }
}