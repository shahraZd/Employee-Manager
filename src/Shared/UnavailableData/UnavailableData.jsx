/**
 *
 * UnavailableData
 *
 */
// @eslint
import React, { memo } from 'react'
import { Link } from 'react-router-dom'
import { Row, Col, Icon, Button } from 'antd'
import { FormattedMessage } from 'react-intl'
import { renderErrorMessage } from './helper'
import messages from './messages'

import './unavailable-data.scss'

function UnavailableData({ iconType, message }) {
  return (
    <div
      className="unavailable-data global-flex-vertical-horizontal-center "
      data-test-id="unreachable-service"
    >
      <Row className="tracks-failure-message ">
        <Col
          className=" unavailable-data-icon "
          sm={24}
          md={24}
          lg={24}
          xl={24}
        >
          <Icon type={iconType} />
        </Col>
        <Col
          sm={24}
          md={24}
          lg={24}
          xl={24}
          className="unavailable-data-message"
        >
          {renderErrorMessage(message)}
        </Col>
        <Col
          sm={24}
          md={24}
          lg={24}
          xl={24}
          className="global-flex-vertical-horizontal-center"
        >
          <Button className="unavailable-data-btn">
            <Link to="/">
              <FormattedMessage {...messages.backDashboard} />
            </Link>
          </Button>
        </Col>
      </Row>
    </div>
  )
}

UnavailableData.propTypes = {}

UnavailableData.defaultProps = {
  iconType: 'info-circle',
  message: <FormattedMessage {...messages.trackFailureMessage} />,
}
export default memo(UnavailableData)
