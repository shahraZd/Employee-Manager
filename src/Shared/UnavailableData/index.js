import UnavailableData from './UnavailableData'

export default UnavailableData
