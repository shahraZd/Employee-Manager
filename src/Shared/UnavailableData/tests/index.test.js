import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { browserHistory, BrowserRouter } from 'react-router-dom'

import UnavailableDataIndex from '../index'
import UnavailableData from '../UnavailableData'
import { DEFAULT_LOCALE } from '../../../../i18n'
import configureStore from '../../../../configureStore'
import 'jest-prop-type-error'

const initialProps = {}

describe('<UnavailableDataIndex />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('Expect to crash if propTypes failed', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <UnavailableDataIndex {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
  })

  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <UnavailableDataIndex {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <UnavailableDataIndex {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })
})

describe('<UnavailableData/>', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <UnavailableData {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
