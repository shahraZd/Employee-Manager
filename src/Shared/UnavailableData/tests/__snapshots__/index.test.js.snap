// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnavailableData/> should render and match the snapshot 1`] = `
<div
  class="unavailable-data global-flex-vertical-horizontal-center "
  data-test-id="unreachable-service"
>
  <div
    class="ant-row tracks-failure-message "
  >
    <div
      class="ant-col  unavailable-data-icon  ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24"
    >
      <i
        aria-label="icon: info-circle"
        class="anticon anticon-info-circle"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="info-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M464 336a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </i>
    </div>
    <div
      class="ant-col unavailable-data-message ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24"
    />
    <div
      class="ant-col global-flex-vertical-horizontal-center ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24"
    >
      <button
        class="ant-btn unavailable-data-btn"
        type="button"
      >
        <a
          href="/"
        >
          <span>
            Back to Dashboard
          </span>
        </a>
      </button>
    </div>
  </div>
</div>
`;
